connector/src/main/java/org/geysermc/connector/health/GeyserHealthProvider.java

```diff
@@ -0,0 +1,150 @@
+    private final GeyserLogger logger;
+
+    public GeyserHealthProvider(GeyserConfiguration.IHealthConfiguration healthConfiguration, GeyserLogger logger) {
+        this.healthConfiguration = healthConfiguration;
+        this.logger = logger;
+    }
+
+    @SuppressWarnings("InfiniteLoopStatement")
+    @Override
+    public void run() {
+        try {
+            ServerSocket socket = new ServerSocket(healthConfiguration.getPort());
+            logger.info("Health provider created.");
+            while (true) {
+                GeyserHelpResponder responder = new GeyserHelpResponder(socket.accept());
+                responder.setDaemon(true);
+                responder.start();
+            }
+        } catch (IOException ex) {
+            logger.error("Failed to initialize server socket.", ex);
+        }
+    }
+
+    private static class GeyserHelpResponder extends Thread {
+        private final Socket inSocket;
+
+        private GeyserHelpResponder(Socket inSocket) {
+            this.inSocket = inSocket;
+        }
+
+        @Override
+        public void run() {
+            try (
+                    InputStreamReader inputStreamReader = new InputStreamReader(inSocket.getInputStream());
+                    BufferedReader bufferedReader = new BufferedReader(inputStreamReader);
+                    OutputStreamWriter outputStreamWriter = new OutputStreamWriter(inSocket.getOutputStream());
+                    PrintWriter printWriter = new PrintWriter(outputStreamWriter)
+            ) {
+                String request = bufferedReader.readLine();
+                if (request == null) return;
+
+                while (true) {
+                    String ignore = bufferedReader.readLine();
+                    if (ignore == null || ignore.length() == 0) break;
+                }
+
+                int initialSplit = request.indexOf(' ');
+                if (initialSplit == -1) {
+                    respondBadRequest(printWriter);
+                    return;
+                }
+
+                String method = request.substring(0, initialSplit);
+
+                if (!method.equalsIgnoreCase("GET")) {
+                    respondNotFound(printWriter);
+                    return;
+                }
+
+                int routeSplit = request.indexOf(' ', initialSplit + 1);
+                if (routeSplit == -1) {
+                    respondBadRequest(printWriter);
+                    return;
+                }
+
+                String route = request.substring(initialSplit + 1, routeSplit);
+                int routeQuerySplit = route.indexOf('?');
+                if (routeQuerySplit != -1) {
+                    route = route.substring(0, routeQuerySplit);
+                }
+
+                if (route.equalsIgnoreCase("/health")) {
+                    respondOk(printWriter);
+                } else {
+                    respondNotFound(printWriter);
```

Comment:
This should be a 503 not 404 as it makes more sense.
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#server_error_responses

> 503 Service Unavailable
The server is not ready to handle the request. Common causes are a server that is down for maintenance or that is overloaded.

>404 Not Found
The server can not find the requested resource. In the browser, this means the URL is not recognized.

Reply:
`this means the URL is not recognized.` Seems like 404 is exactly what we're after since the route could not be found / retrieved. A 503 would mean we can't handle the response but we can, and are.

Reply:
I agree with Corey's interpretation on this.

Reply:
I still feel like 404 is the wrong code, since technically that route exists...
Wait, I see what's happening now. I should have read the if condition first 😅